lib/db/mysql.js

```diff
@@ -566,21 +566,31 @@ module.exports = function (log, error) {
 
   // Internal
 
-  MySql.prototype.singleQuery = function (poolName, sql, params) {
+  MySql.prototype.singleQuery = function (poolName, sql, params, readUncommitted) {
     return this.getConnection(poolName)
       .then(
         function (connection) {
-          return query(connection, sql, params)
-            .then(
-              function (result) {
-                connection.release()
-                return result
-              },
-              function (err) {
-                connection.release()
-                throw err
-              }
-            )
+          if (readUncommitted) {
+            return query(connection, 'SET TRANSACTION ISOLATION LEVEL READ UNCOMMITTED')
```

Comment:
Careful, you'll have to make sure you undo this after running the statement, or the connection will be returned to the pool with the isolation level still at `READ UNCOMMITTED`.  Not a problem in this case since the script will be running in a separate process, but a potential footgun for the future.